src/search/search_engines/eager_search.cc

```diff
@@ -33,6 +38,15 @@ EagerSearch::EagerSearch(const Options &opts)
         cerr << "lazy_evaluator must cache its estimates" << endl;
         utils::exit_with(utils::ExitCode::SEARCH_INPUT_ERROR);
     }
+    // Even for GBFS, we need a g-evaluator for reporting progress.
+    if (!g_evaluator) {
+        shared_ptr<AbstractTask> task = tasks::g_root_task;
+        OperatorCost cost_type = opts.get<OperatorCost>("cost_type");
+        if (cost_type != OperatorCost::NORMAL) {
+            task = make_shared<tasks::CostAdaptedTask>(task, cost_type);
+        }
+        g_evaluator = make_shared<g_evaluator::GEvaluator>(task);
```

Comment:
I agree, this is definitely problematic in the current patch. I see two options: 1) we remove cost_type from the options and expose g_eval on the commandline (breaking backwards compatibility of the configs). 2) we make sure that users can never pass cost_type and g_eval at the same time. I tried to use 2) in the patch, but I forgot to use the cost_type for creating the g_eval. I'll fix that. I'd like to avoid 1) to avoid breaking old configs, but maybe we'll see later that we have to break them.

Reply:
We decided to switch to option 1).